python/lsst/afw/coord/coord.cc

```diff
@@ -0,0 +1,300 @@
+    clsCoord.def("toEcliptic", (EclipticCoord (Coord::*)(double const) const) &Coord::toEcliptic);
+    clsCoord.def("toEcliptic", (EclipticCoord (Coord::*)() const) &Coord::toEcliptic);
+    clsCoord.def("toTopocentric", &Coord::toTopocentric);
+
+    py::class_<IcrsCoord, std::shared_ptr<IcrsCoord>> clsIcrsCoord(mod, "IcrsCoord", py::base<Coord>());
+
+    /* Constructors */
+    clsIcrsCoord.def(py::init<lsst::afw::geom::Point2D const &, lsst::afw::geom::AngleUnit>(),
+        py::arg("p2d"), py::arg("unit") = lsst::afw::geom::degrees);
+    clsIcrsCoord.def(py::init<lsst::afw::geom::Point3D const &, bool, lsst::afw::geom::Angle const>(),
+        py::arg("p3d"), py::arg("normalize") = true, py::arg("defaultLongitude") = lsst::afw::geom::Angle(0.));
+    clsIcrsCoord.def(py::init<lsst::afw::geom::Angle const, lsst::afw::geom::Angle const>(),
+        py::arg("ra"), py::arg("dec"));
+    clsIcrsCoord.def(py::init<std::string const, std::string const>(),
+        py::arg("ra"), py::arg("dec"));
+    clsIcrsCoord.def(py::init<>());
+
+    /* Operators */
+    clsIcrsCoord.def("__str__", [](IcrsCoord &c) {
+        std::ostringstream os;
+        os << c;
+        return os.str();
+    });
+
+    /* Members */
+    clsIcrsCoord.def("clone", &IcrsCoord::clone);
+    clsIcrsCoord.def("getClassName", &IcrsCoord::getClassName);
+    clsIcrsCoord.def("getCoordSystem", &IcrsCoord::getCoordSystem);
+    clsIcrsCoord.def("reset", &IcrsCoord::reset);
+    clsIcrsCoord.def("getRa", &IcrsCoord::getRa);
+    clsIcrsCoord.def("getDec", &IcrsCoord::getDec);
+    clsIcrsCoord.def("getRaStr", &IcrsCoord::getRaStr);
+    clsIcrsCoord.def("getDecStr", &IcrsCoord::getDecStr);
+    clsIcrsCoord.def("toFk5", (Fk5Coord (IcrsCoord::*)(double const) const) &IcrsCoord::toFk5);
+    clsIcrsCoord.def("toFk5", (Fk5Coord (IcrsCoord::*)() const) &IcrsCoord::toFk5);
+    clsIcrsCoord.def("toIcrs", &IcrsCoord::toIcrs);
+
+    py::class_<Fk5Coord, std::shared_ptr<Fk5Coord>> clsFk5Coord(mod, "Fk5Coord", py::base<Coord>());
+
+    /* Constructors */
+    clsFk5Coord.def(py::init<lsst::afw::geom::Point2D const &, lsst::afw::geom::AngleUnit, double const>(),
+        py::arg("p2d"), py::arg("unit") = lsst::afw::geom::degrees, py::arg("epoch") = 2000.0);
+    clsFk5Coord.def(py::init<lsst::afw::geom::Point3D const &, double const, bool, lsst::afw::geom::Angle const>(),
+        py::arg("p3d"), py::arg("epoch") = 2000.0, py::arg("normalize") = true, py::arg("defaultLongitude") = lsst::afw::geom::Angle(0.));
+    clsFk5Coord.def(py::init<lsst::afw::geom::Angle const, lsst::afw::geom::Angle const, double const>(),
+        py::arg("ra"), py::arg("dec"), py::arg("epoch") = 2000.0);
+    clsFk5Coord.def(py::init<std::string const, std::string const, double const>(),
+        py::arg("ra"), py::arg("dec"), py::arg("epoch") = 2000.0);
+    clsFk5Coord.def(py::init<>());
+
+    /* Operators */
+    clsFk5Coord.def("__str__", [](Fk5Coord &c) {
```

Comment:
maybe we should add a `toString` method to Coord like most other afw classes have and then this would be less repetitive.